py3status/modules/whatismyip.py

```diff
@@ -61,43 +74,38 @@ def on_click(self, event):
         else:
             self.mode = 'ip'
 
-    def _get_my_ip_and_location(self):
+    def _get_my_ip_info(self):
         """
         """
         try:
-            if self.py3.format_contains(self.format, 'country'):
-                resp = urlopen(self.url_geo, timeout=self.timeout).read()
-                resp = json.loads(resp)
-                country = resp['country']
-                ip = resp['query']
-            else:
-                country = None
-                ip = urlopen(self.url, timeout=self.timeout).read()
-                ip = ip.decode('utf-8')
+            resp = self.py3.request(self.url_geo, timeout=self.timeout).json()
+            info = {}
+            for placeholder in self.py3.get_placeholders_list(self.format):
+                if placeholder in resp:
+                    info[placeholder] = resp[placeholder]
+            for old, new in self.substitutions.items():
+                info[old] = resp.get(new)
+            return info
         except Exception:
-            country = None
-            ip = None
-        return country, ip
+            return None
```

Comment:
No idea if we want this or to use `except self.Py3.RequestException:` instead.... or to use `except (self.py3.RequestException):` like bunch of other modules. My noob review here is done and @tobes said he went out in other issue... just a heads up. Fix exception bit and we'll get this in probably tomorrow. 👍